evalml/tests/automl_tests/test_automl_search_classification.py

```diff
@@ -715,3 +715,61 @@ def test_tuning_threshold_objective(mock_predict, mock_fit, mock_score, mock_enc
         assert automl.best_pipeline.threshold is None
     else:
         assert automl.best_pipeline.threshold == 0.6
+
+
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass'])
+@pytest.mark.parametrize("categorical_features", ['none', 'some', 'all'])
+@pytest.mark.parametrize("size", ['small', 'large'])
+@pytest.mark.parametrize("sampling_ratio", [0.8, 0.5, 0.25, 0.2, 0.1, 0.05])
+def test_automl_search_sampler_ratio(sampling_ratio, size, categorical_features, problem_type, mock_imbalanced_data_X_y, has_minimal_dependencies):
+    X, y = mock_imbalanced_data_X_y(problem_type, categorical_features, size)
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=problem_type, sampler_method='auto', sampler_balanced_ratio=sampling_ratio)
+    pipelines = automl.allowed_pipelines
+    if sampling_ratio <= 0.2:
+        # we consider this balanced, so we expect no samplers
+        assert not any(any("sampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+    else:
+
+        assert all(any("Undersampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+
+
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass'])
+@pytest.mark.parametrize("sampler_method,categorical_features", [(None, 'none'), (None, 'some'), (None, 'all'),
+                                                                 ('Undersampler', 'none'), ('Undersampler', 'some'), ('Undersampler', 'all')])
+def test_automl_search_sampler_method(sampler_method, categorical_features, problem_type, mock_imbalanced_data_X_y, has_minimal_dependencies, caplog):
+    # 0.2 minority:majority class ratios
+    X, y = mock_imbalanced_data_X_y(problem_type, categorical_features, 'small')
+    automl = AutoMLSearch(X_train=X, y_train=y, problem_type=problem_type, sampler_method=sampler_method)
+    # since our default sampler_balanced_ratio for AutoMLSearch is 0.25, we should be adding the samplers when we can
+    pipelines = automl.allowed_pipelines
+    if sampler_method is None:
+        assert not any(any("sampler" in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+    else:
+        assert all(any(sampler_method in comp.name for comp in pipeline.component_graph) for pipeline in pipelines)
+
+
+@pytest.mark.parametrize("sampler_methods", [None, 'Undersampler'])
+@pytest.mark.parametrize("problem_type", ['binary', 'multiclass'])
+@patch("evalml.pipelines.components.estimators.Estimator.fit")
+@patch('evalml.pipelines.BinaryClassificationPipeline.score', return_value={"Log Loss Binary": 0.5})
+@patch('evalml.pipelines.MulticlassClassificationPipeline.score', return_value={"Log Loss Multiclass": 0.5})
+def test_oversampling_in_pipelines(multiclass_score, binary_score, mock_fit, problem_type, sampler_methods, has_minimal_dependencies):
```

Comment:
Rename to `test_sampling_in_pipelines` yes?

Reply:
Actually, I think we should delete this test.

Reasoning: `test_automl_search_sampler_ratio` and `test_automl_search_sampler_method` verify that the sampler is or is not included in the pipelines. And from the perspective of test coverage for `AutoMLSearch`, that is sufficient. We can then write separate pipeline or component tests to verify that the undersampler component behaves as we expect it to. But I don't think we should mix the two.